tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -133,9 +115,10 @@ def check(self, _):
 
         self.log.debug("Connecting to %s on port %d", self.host, self.port)
 
-        for addr in self.addrs:
+        for addr, socket_type in self.addrs:
+            self.socket_type = socket_type
```

Comment:
I don't think we need `self.socket` anymore, since its use is fully confined to this scope.